pyprof/prof/blas.py

```diff
@@ -132,7 +132,11 @@ def __init__(self, d):
         assert (k1 == k2)
         t1 = A['dtype']
         t2 = B['dtype']
-        assert (t1 == t2)
+        # assert (t1 == t2).  No longer true.
+        if t1 != t2:
+            if t1 == 'float16' or t2 == 'float16' :
+                A['dtype'] = 'float16'
+                B['dtype'] = 'float16'
```

Comment:
Can you provide a simple test case?

Reply:
If this is true, it will impact all other matmul operations as well, not just bmm.

Reply:
Also, can you provide a link to documentation that indicates this is not true? The comment is both obvious (i.e. "assert (t1 == t2)" and the code `t1 != t2`, the comment provide no additional insight). But "No longer true" is vague. Why is this no longer true?